src/colvarcomp_distances.cpp

```diff
@@ -1045,6 +1045,44 @@ colvar::rmsd::rmsd(std::string const &conf)
     // this is only required for ABF, but we do both groups here for better caching
     atoms->rot.request_group2_gradients(atoms->size());
   }
+
+  std::string perm_conf;
+  size_t pos = 0; // current position in config string
+  n_permutations = 1;
+
+  while (key_lookup(conf, "atomPermutation", &perm_conf, &pos)) {
+    std::vector<size_t> perm;
+    if (perm_conf.size()) {
+      std::istringstream is(perm_conf);
+      size_t index;
+      while (is >> index) {
+        std::vector<int> const &ids = atoms->ids();
+        size_t const ia = std::find(ids.begin(), ids.end(), index-1) - ids.begin();
```

Comment:
@giacomofiorin I have a slight doubt about this: are we sure that the internal atom ids are just the user-facing 1-based numbers minus 1? If not, how do we find one given the other? Should the proxy report if the back-end's internal numbers are 1-based or 0-based, or provide a mapping function? Specifically, I'm thinking of the upcoming Fortran-based back-ends.

Reply:
There is no guarantee, although in all backends this is true at the moment because atom IDs are just array indices (0-based). What I find tricky with this approach is that you could select atoms in a compact way (e.g. with `atomNameResidueRange`) but you would need then to use internal indices anyway to write the permutation. It would seem clearer to me to specify each permutation as its own selection and reference coordinates (see my other comment), unless I'm missing something.

Reply:
For the permutation you need to mention atoms individually, but I agree that it would be nice to skip atoms that are unchanged.